common/src/Renderer/EntityDecalRenderer.h

```diff
@@ -0,0 +1,157 @@
+/*
+ Copyright (C) 2023 Kristian Duske
+
+ This file is part of TrenchBroom.
+
+ TrenchBroom is free software: you can redistribute it and/or modify
+ it under the terms of the GNU General Public License as published by
+ the Free Software Foundation, either version 3 of the License, or
+ (at your option) any later version.
+
+ TrenchBroom is distributed in the hope that it will be useful,
+ but WITHOUT ANY WARRANTY; without even the implied warranty of
+ MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ GNU General Public License for more details.
+
+ You should have received a copy of the GNU General Public License
+ along with TrenchBroom. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#pragma once
+
+#include "AllocationTracker.h"
+#include "Color.h"
+#include "Renderer/EdgeRenderer.h"
+#include "Renderer/EntityModelRenderer.h"
+#include "Renderer/FaceRenderer.h"
+#include "Renderer/Renderable.h"
+
+#include <kdl/vector_set.h>
+
+#include <memory>
+#include <unordered_map>
+#include <unordered_set>
+#include <vector>
+
+namespace TrenchBroom
+{
+namespace Assets
+{
+class EntityModelManager;
+struct DecalSpecification;
+} // namespace Assets
+
+namespace Model
+{
+class BrushFace;
+class Node;
+class BrushNode;
+class EditorContext;
+class EntityNode;
+} // namespace Model
+
+namespace View
+{
+class MapDocument; // FIXME: Renderer should not depend on View
+} // namespace View
+
+namespace Renderer
+{
+class AttrString;
+
+class EntityDecalRenderer
+{
+private:
+  struct EntityDecalData
+  {
+    std::unordered_set<const Model::BrushNode*> brushes;
```

Comment:
In C++, it is often so that `std::vector` with linear search is faster than data structures like `std::unordered_set` and such because of CPU caching. Unless you are dealing with large datasets, you can usually just go with a vector.